lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -2428,6 +2428,86 @@ void BoundInterpreterFunction::fwdBatchedReduceAddInst(
                             eBatchDims, eDestDims);
 }
 
+template <typename ElemTy>
+void BoundInterpreterFunction::fwdBatchedReduceMinInstImpl(
+    Value *batch, Value *dest, const ShapeVector &eBatchDims,
+    const ShapeVector &eDestDims, ElemTy max) {
+  static_assert(max_tensor_dimensions == 6,
+                "Loops below assume max_tensor_dimensions = 6.");
+  // Get unowned handles of the batch and dest with these new expanded dims.
+  auto eBatch = getTensor(batch)->getUnowned(eBatchDims);
+  auto eDest = getTensor(dest)->getUnowned(eDestDims);
+  auto eBatchH = eBatch.getHandle<ElemTy>();
+  auto eDestH = eDest.getHandle<ElemTy>();
+  eDestH.clear(max);
+
+  unsigned int axes[max_tensor_dimensions];
+  for (int i = 0; i < max_tensor_dimensions; i++) {
+    axes[i] = (eDestDims[i] > 1);
+  }
+
+  // We can use this loop for all shapes. Use the same indices for both the
+  // batch and dest, except for setting the axis index in the dest to 0.
+  for (size_t x = 0, dx = 0; x < eBatchDims[0]; x++, dx += axes[0]) {
+    for (size_t y = 0, dy = 0; y < eBatchDims[1]; y++, dy += axes[1]) {
+      for (size_t z = 0, dz = 0; z < eBatchDims[2]; z++, dz += axes[2]) {
+        for (size_t w = 0, dw = 0; w < eBatchDims[3]; w++, dw += axes[3]) {
+          for (size_t q = 0, dq = 0; q < eBatchDims[4]; q++, dq += axes[4]) {
+            for (size_t r = 0, dr = 0; r < eBatchDims[5]; r++, dr += axes[5]) {
+              size_t destIndices[] = {dx, dy, dz, dw, dq, dr};
+              size_t srcIndices[] = {x, y, z, w, q, r};
+              eDestH.at(destIndices) =
+                  eDestH.at(destIndices) < eBatchH.at(srcIndices)
+                      ? eDestH.at(destIndices)
+                      : eBatchH.at(srcIndices);
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+void BoundInterpreterFunction::fwdBatchedReduceMinInst(
+    const glow::BatchedReduceMinInst *I) {
+
+  auto *batch = I->getBatch();
+  auto *dest = I->getDest();
+  const auto axes = I->getAxes();
+
+  // Initialize both expanded batch and dest dims to the expanded batch
+  // dims. This allows us below to iterate over the tensor regardless of its
+  // shape using max_tensor_dimensions loops below.
+  ShapeVector eBatchDims = expandDimsToMax(batch->dims());
+  ShapeVector eDestDims = eBatchDims;
+  // Set the destination axes dimensions (the one we are reducing) to 1.
+  for (int i = 0; i < axes.size(); i++) {
+    eDestDims[axes[i]] = 1;
+  }
+
+  switch (batch->getElementType()) {
```

Comment:
can you reuse: dispatchFloatingPointImpl and dispatchIndexTypeImpl here?

Reply:
I can maybe use dispatchArithmeticImpl which supports all the four types I need. Will that be ok?